content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+
+```java
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@SpringBatchTest
+public class BillrunApplicationTests {
+
+	@Autowired
+	private JobLauncherTestUtils jobLauncherTestUtils;
+
+	@Autowired
+	private DataSource dataSource;
+
+	private JdbcTemplate jdbcTemplate;
+
+	@Before
+	public void setup()  {
+		this.jdbcTemplate = new JdbcTemplate(this.dataSource);
+	}
+
+	@Test
+	public void testJobResults() throws Exception{
+		testResult();
+	}
+
+
+	private void testResult() {
+		List<BillStatement> billStatements = this.jdbcTemplate.query("select ID, " +
+				"first_name, last_name, minutes, data_usage, bill_amount FROM " +
+				"bill_statements",
+				(rs, rowNum) -> new BillStatement(rs.getLong("id"),
+						rs.getString("FIRST_NAME"), rs.getString("LAST_NAME"),
+						rs.getLong("MINUTES"), rs.getLong("DATA_USAGE"),
+						rs.getDouble("bill_amount")));
+		assertThat(billStatements.size()).isEqualTo(5);
+
+		BillStatement billStatement = billStatements.get(0);
+		assertThat(billStatement.getBillAmount()).isEqualTo(6);
+		assertThat(billStatement.getFirstName()).isEqualTo("jane");
+		assertThat(billStatement.getLastName()).isEqualTo("doe");
+		assertThat(billStatement.getId()).isEqualTo(1);
+		assertThat(billStatement.getMinutes()).isEqualTo(500);
+		assertThat(billStatement.getDataUsage()).isEqualTo(1000);
+
+	}
+
+	public static class BillStatement extends Usage {
+
+		public BillStatement(Long id, String firstName, String lastName, Long minutes, Long dataUsage, double billAmount) {
+			super(id, firstName, lastName, minutes, dataUsage);
+			this.billAmount = billAmount;
+		}
+
+		private double billAmount;
+
+		public double getBillAmount() {
+			return billAmount;
+		}
+
+		public void setBillAmount(double billAmount) {
+			this.billAmount = billAmount;
+		}
+	}
+
+}
+
+```
 
 ## Deployment
 
 Deploy to local, Cloud Foundry and Kubernetes
 
 ### Local
 
-get the jar run it bla
+1.  Now let’s take the next step of building the project.
+    From a command line change directory to the location of your project and build the project using maven:
+    `mvn clean package`.
+
+2.  Now let’s execute the application with the configurations required to process the usage information in the database.
+
+    To configure the execution of the billrun application utilize the following arguments:
+
+    1. _spring.datasource.url_ - set the URL to your database instance. In the sample below we are connecting to a mysql `task` database on our local machine at port 3306.
+    1. _spring.datasource.username_ - the user name to be used for the MySql database. In the sample below it is `root`
+    1. _spring.datasource.password_ - the password to be used for the MySql database. In the sample below it is `password`
+    1. _spring.datasource.driverClassName_ - The driver to use to connect to the MySql database. In the sample below it is `com.mysql.jdbc.Driver'
```

Comment:
The char at the end of this line is a single quote instead of a back tic so it breaks the formatting.

Reply:
done